PoseEstimation-TFLiteSwift/OpenPose/OpenPosePoseEstimator.swift

```diff
@@ -193,29 +269,161 @@ private extension PoseEstimationOutput {
             }
             return Human(keypoints: keypoints, lines: [])
         }
+    }
+    
+    func parseAllPartOnMultiHuman(from output: TFLiteFlatArray<Float32>, with threshold: Float?) -> [Human] {
+        let parts = OpenPosePoseEstimator.Output.BodyPart.allCases
+        var verticesForEachPart: [[(col: Int, row: Int, val: Float32)]?] = parts.map { _ in nil }
+        let pairs = OpenPosePoseEstimator.Output.BodyPart.lines
+        var edgesForEachPair: [[(from: (col: Int, row: Int, val: Float32), to: (col: Int, row: Int, val: Float32), cost: Float32)]] = pairs.map { _ in [] }
+        let (colSize, rowSize) = (OpenPosePoseEstimator.Output.ConfidenceMap.width,
+                                  OpenPosePoseEstimator.Output.ConfidenceMap.height)
         
-        // 2.
+        let pairCount = pairs.count
+        for (pairIndex, pair) in pairs.enumerated() {
+            // guard pairIndex == 0 else { continue }
+            
+            let startingPartIndex = pair.from.offsetValue()
+            let endingPartIndex = pair.to.offsetValue()
+            
+            let thresholdRatio: Float = 0.5 + (1.0 - (Float(pairIndex+1)/Float(pairCount))) * 0.5 // 1.0~>0.5
+            
+            // 1. Non Maximum Suppression, 2. Create Bipartite Graph
+            if verticesForEachPart[startingPartIndex] == nil {
+                verticesForEachPart[startingPartIndex] = output.keypoints(partIndex: startingPartIndex, filterSize: 5, threshold: threshold*thresholdRatio)
+            }
+            
+            // 1. Non Maximum Suppression, 2. Create Bipartite Graph
+            if verticesForEachPart[endingPartIndex] == nil {
+                verticesForEachPart[endingPartIndex] = output.keypoints(partIndex: endingPartIndex, filterSize: 5, threshold: threshold*thresholdRatio)
+            }
+            
+            guard let startingPartVertices = verticesForEachPart[startingPartIndex],
+                let endingPartVertices = verticesForEachPart[endingPartIndex] else { fatalError("cannot get verties for assignment") }
+            for startingPartVertex in startingPartVertices {
+                for endingPartVertex in endingPartVertices {
+                    // 3. Line Integral
+                    // vx, vy
+                    let (sCol, eCol) = (startingPartVertex.col, endingPartVertex.col)
+                    let (sRow, eRow) = (startingPartVertex.row, endingPartVertex.row)
+                    let (xDiff, yDiff) = (Float(eCol-sCol), Float(eRow-sRow))
+                    guard xDiff != 0 || yDiff != 0 else { continue }
+                    let vLength = sqrt(pow(xDiff, 2) + pow(yDiff, 2))
+                    let (vx, vy) = (xDiff/vLength, yDiff/vLength)
+                    // sampling
+                    let numberOfSamples = 10
+                    let dx = Float(xDiff) / Float(numberOfSamples)
+                    let dy = Float(yDiff) / Float(numberOfSamples)
```

Comment:
are types of xDiff, yDiff Float?

Reply:
@syjdev 
Oh, that's right. There is no need to cast.